### PR TITLE
Create 24_conditional-workflow.md

### PR DESCRIPTION
**Pull request by jjkoehorst**
Initial version for conditional workflows.

### CODE REVIEW

File: _episodes/24_conditional-workflow.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  step1:
+    in:
+      in1: val
+      a_new_var: val
+    run: foo.cwl
+    when: $(inputs.in1 < 1)
+    out: [out1]
+```
+
+Using the following command `cwltool cond-wf-003.1.cwl --val 0` the value will pass the first conditional step and will therefor be executed and is shown in the log by `INFO [step step1] start` whereas the second step is skipped as indicated by `INFO [step step2] will be skipped`.
```
Comment (kaushik-work) [inline _episodes/24_conditional-workflow.md:81]: Worthwhile to note here that we can have variables in the expression that are not present in the underlying process. In this case it is `a_new_var`

Comment (kaushik-work) [reply _episodes/24_conditional-workflow.md:81]: Might be more instructive to actually use `a_new_var` in the expression as well or instead of `in1`

**Review by kaushik-work — DISMISSED** [review]
This is great. Thank you! A follow up section indicating the use of `pickValue` to implement the patterns outlined in the proposal https://github.com/common-workflow-language/common-workflow-language/issues/854 may be helpful.

File: _episodes/24_conditional-workflow.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      - step1/out1
+      - step2/out1
+    pickValue: first_non_null
+
+requirements:
+  InlineJavascriptRequirement: {}
+  MultipleInputFeatureRequirement: {}
+```
+
+The first thing you'll notice is that this workflow is only compatible for cwl 1.2+
```
Comment (mr-c) [inline _episodes/24_conditional-workflow.md:60]:
```suggestion
The first thing you'll notice is that this workflow is only compatible for version 1.2 or greater of the CWL standards.
```

**Review by mr-c — APPROVED** [review]

**Review by mr-c — APPROVED** [review]

**Pull request merged**